test/e2e-go/restAPI/restClient_test.go

```diff
@@ -1227,3 +1227,168 @@ func TestNilStateProofInParticipationInfo(t *testing.T) {
+    box_create              // [] // boxes: arg[1] -> [5]byte
+    b end
+del:                        // delete box arg[1]
+    txn ApplicationArgs 0   // [arg[0]]
+    byte "delete"           // [arg[0], "delete"]
+    ==                      // [arg[0]=?="delete"]
+    bz bad                  // "delete" ? continue : goto bad
+    txn ApplicationArgs 1   // [arg[1]]
+    box_del                 // del boxes[arg[1]]
+    b end
+bad:
+    err
+end:
+    int 1
+`
+	ops, err := logic.AssembleString(prog)
+	approval := ops.Program
+	ops, err = logic.AssembleString("#pragma version 7\nint 1")
+	clearState := ops.Program
+
+	gl := basics.StateSchema{}
+	lc := basics.StateSchema{}
+
+	// create app
+	appCreateTxn, err := testClient.MakeUnsignedApplicationCallTx(
+		0, nil, nil, nil,
+		nil, nil, transactions.NoOpOC,
+		approval, clearState, gl, lc, 0,
+	)
+	a.NoError(err)
+	appCreateTxn, err = testClient.FillUnsignedTxTemplate(someAddress, 0, 0, 0, appCreateTxn)
+	a.NoError(err)
+	appCreateTxID, err := testClient.SignAndBroadcastTransaction(wh, nil, appCreateTxn)
+	a.NoError(err)
+	_, err = waitForTransaction(t, testClient, someAddress, appCreateTxID, 30*time.Second)
+	a.NoError(err)
+
+	// get app ID
+	submittedAppCreateTxn, err := testClient.PendingTransactionInformationV2(appCreateTxID)
+	a.NoError(err)
+	a.NotNil(submittedAppCreateTxn.ApplicationIndex)
+	createdAppID := basics.AppIndex(*submittedAppCreateTxn.ApplicationIndex)
+	a.Greater(uint64(createdAppID), uint64(0))
+
+	// fund app account
+	appFundTxn, err := testClient.SendPaymentFromWallet(
+		wh, nil, someAddress, createdAppID.Address().String(),
+		0, 10_000_000, nil, "", 0, 0,
+	)
+	a.NoError(err)
+	appFundTxID := appFundTxn.ID()
+	_, err = waitForTransaction(t, testClient, someAddress, appFundTxID.String(), 30*time.Second)
+	a.NoError(err)
+
+	// define operate box helper
+	operateBoxAndSendTxn := func(operation, boxName string) (txID string, err error) {
+		tx, err := testClient.MakeUnsignedAppNoOpTx(
+			uint64(createdAppID),
+			[][]byte{[]byte(operation), []byte(boxName)},
+			nil, nil, nil,
+			[]transactions.BoxRef{
+				{
+					Name: []byte(boxName), Index: 0,
+				},
+			},
+		)
+		if err != nil {
+			return
+		}
+		tx, err = testClient.FillUnsignedTxTemplate(someAddress, 0, 0, 0, tx)
+		if err != nil {
+			return
+		}
+		wh, err = testClient.GetUnencryptedWalletHandle()
+		if err != nil {
+			return
+		}
+		txID, err = testClient.SignAndBroadcastTransaction(wh, nil, tx)
+		if err != nil {
+			return
+		}
+		_, err = waitForTransaction(t, testClient, someAddress, txID, 30*time.Second)
+		// no matter wait for txn returns error or not, we should all return with txID and err
+		return
+	}
+
+	// iterate and create boxes
+	totalBoxNum := 10
+
+	resp, err := testClient.ApplicationBoxes(uint64(createdAppID))
+	a.NoError(err)
+	a.Empty(resp.Boxes)
+
+	createdBoxName := map[string]bool{}
+	createdBoxCount := 0
+
+	for boxIterIndex := 0; boxIterIndex < totalBoxNum; boxIterIndex++ {
```

Comment:
I'd like to see unusual cases tested here, rather than just "box1" "box2" etc.

Try names with 0 bytes. Trying names with weird unicode. Try names with with troublesome http and/or db characters like quotes, spaces, =, &, etc. Write a helper function so it's very obvious which ones you try (rather than a very, very long test with a ton of boilerplate)